File: game.cpp
/*
 * XO Game (Tic-Tac-Toe) - C++ Implementation
 * لعبة إكس أو - نسخة C++
 * 
 * هذا الملف يحتوي على منطق اللعبة الكامل بلغة C++
 * يمكن استخدامه كمرجع أو دمجه مع واجهات رسومية مختلفة
 */

#include <iostream>
#include <vector>
#include <algorithm>
#include <limits>
#include <ctime>
#include <cstdlib>

using namespace std;

// ===== Constants =====
const int BOARD_SIZE = 9;
const char EMPTY = ' ';
const char PLAYER_X = 'X';
const char PLAYER_O = 'O';

// ===== Winning Combinations =====
const int WINNING_COMBINATIONS[8][3] = {
    {0, 1, 2}, {3, 4, 5}, {6, 7, 8}, // Rows
    {0, 3, 6}, {1, 4, 7}, {2, 5, 8}, // Columns
    {0, 4, 8}, {2, 4, 6}             // Diagonals
};

// ===== Game Class =====
class TicTacToe {
private:
    vector<char> board;
    char currentPlayer;
    char humanPlayer;
    char aiPlayer;
    int difficulty; // 1=Easy, 2=Medium, 3=Hard
    int scoreX, scoreO, scoreDraw;
    bool gameOver;
    
public:
    // Constructor
    TicTacToe() {
        board.resize(BOARD_SIZE, EMPTY);
        currentPlayer = PLAYER_X;
        humanPlayer = PLAYER_X;
        aiPlayer = PLAYER_O;
        difficulty = 2; // Medium by default
        scoreX = 0;
        scoreO = 0;
        scoreDraw = 0;
        gameOver = false;
        srand(time(0));
    }
    
    // Display Board
    void displayBoard() {
        cout << "\n";
        cout << "  " << board[0] << " | " << board[1] << " | " << board[2] << "\n";
        cout << " -----------\n";
        cout << "  " << board[3] << " | " << board[4] << " | " << board[5] << "\n";
        cout << " -----------\n";
        cout << "  " << board[6] << " | " << board[7] << " | " << board[8] << "\n";
        cout << "\n";
    }
    
    // Check if cell is empty
    bool isCellEmpty(int index) {
        return board[index] == EMPTY;
    }
    
    // Make move
    bool makeMove(int index, char player) {
        if (index < 0 || index >= BOARD_SIZE || !isCellEmpty(index)) {
            return false;
        }
        board[index] = player;
        return true;
    }
    
    // Check winner
    char checkWinner() {
        // Check all winning combinations
        for (int i = 0; i < 8; i++) {
            int a = WINNING_COMBINATIONS[i][0];
            int b = WINNING_COMBINATIONS[i][1];
            int c = WINNING_COMBINATIONS[i][2];
            
            if (board[a] != EMPTY && 
                board[a] == board[b] && 
                board[a] == board[c]) {
                return board[a];
            }
        }
        
        // Check for draw
        bool isFull = true;
        for (int i = 0; i < BOARD_SIZE; i++) {
            if (board[i] == EMPTY) {
                isFull = false;
                break;
            }
        }
        
        if (isFull) return 'D'; // Draw
        
        return EMPTY; // Game continues
    }
    
    // Get available moves
    vector<int> getAvailableMoves() {
        vector<int> moves;
        for (int i = 0; i < BOARD_SIZE; i++) {
            if (board[i] == EMPTY) {
                moves.push_back(i);
            }
        }
        return moves;
    }
    
    // Random move (Easy AI)
    int getRandomMove() {
        vector<int> moves = getAvailableMoves();
        if (moves.empty()) return -1;
        return moves[rand() % moves.size()];
    }
    
    // Minimax Algorithm with Alpha-Beta Pruning (Hard AI)
    int minimax(vector<char>& tempBoard, int depth, bool isMaximizing, int alpha, int beta) {
        char winner = checkWinnerForBoard(tempBoard);
        
        if (winner == aiPlayer) return 10 - depth;
        if (winner == humanPlayer) return depth - 10;
        if (winner == 'D') return 0;
        
        if (isMaximizing) {
            int bestScore = numeric_limits<int>::min();
            for (int i = 0; i < BOARD_SIZE; i++) {
                if (tempBoard[i] == EMPTY) {
                    tempBoard[i] = aiPlayer;
                    int score = minimax(tempBoard, depth + 1, false, alpha, beta);
                    tempBoard[i] = EMPTY;
                    bestScore = max(score, bestScore);
                    alpha = max(alpha, bestScore);
                    if (beta <= alpha) break; // Alpha-Beta Pruning
                }
            }
            return bestScore;
        } else {
            int bestScore = numeric_limits<int>::max();
            for (int i = 0; i < BOARD_SIZE; i++) {
                if (tempBoard[i] == EMPTY) {
                    tempBoard[i] = humanPlayer;
                    int score = minimax(tempBoard, depth + 1, true, alpha, beta);
                    tempBoard[i] = EMPTY;
                    bestScore = min(score, bestScore);
                    beta = min(beta, bestScore);
                    if (beta <= alpha) break; // Alpha-Beta Pruning
                }
            }
            return bestScore;
        }
    }
    
    // Check winner for temporary board
    char checkWinnerForBoard(vector<char>& tempBoard) {
        for (int i = 0; i < 8; i++) {
            int a = WINNING_COMBINATIONS[i][0];
            int b = WINNING_COMBINATIONS[i][1];
            int c = WINNING_COMBINATIONS[i][2];
            
            if (tempBoard[a] != EMPTY && 
                tempBoard[a] == tempBoard[b] && 
                tempBoard[a] == tempBoard[c]) {
                return tempBoard[a];
            }
        }
        
        bool isFull = true;
        for (int i = 0; i < BOARD_SIZE; i++) {
            if (tempBoard[i] == EMPTY) {
                isFull = false;
                break;
            }
        }
        
        if (isFull) return 'D';
        return EMPTY;
    }
    
    // Get best move using Minimax
    int getBestMove() {
        int bestScore = numeric_limits<int>::min();
        int bestMove = -1;
        
        for (int i = 0; i < BOARD_SIZE; i++) {
            if (board[i] == EMPTY) {
                vector<char> tempBoard = board;
                tempBoard[i] = aiPlayer;
                int score = minimax(tempBoard, 0, false, 
                                   numeric_limits<int>::min(), 
                                   numeric_limits<int>::max());
                
                if (score > bestScore) {
                    bestScore = score;
                    bestMove = i;
                }
            }
        }
        
        return bestMove;
    }
    
    // AI Move
    int getAIMove() {
        switch(difficulty) {
            case 1: // Easy
                return getRandomMove();
            case 2: // Medium
                return (rand() % 2 == 0) ? getBestMove() : getRandomMove();
            case 3: // Hard
                return getBestMove();
            default:
                return getRandomMove();
        }
    }
    
    // Reset game
    void resetGame() {
        fill(board.begin(), board.end(), EMPTY);
        currentPlayer = PLAYER_X;
        gameOver = false;
    }
    
    // Reset scores
    void resetScores() {
        scoreX = 0;
        scoreO = 0;
        scoreDraw = 0;
    }
    
    // Display scores
    void displayScores() {
        cout << "\n===== النتائج =====\n";
        cout << "X: " << scoreX << " | ";
        cout << "تعادل: " << scoreDraw << " | ";
        cout << "O: " << scoreO << "\n";
        cout << "==================\n";
    }
    
    // Set difficulty
    void setDifficulty(int level) {
        if (level >= 1 && level <= 3) {
            difficulty = level;
        }
    }
    
    // Set player choice
    void setHumanPlayer(char player) {
        if (player == PLAYER_X || player == PLAYER_O) {
            humanPlayer = player;
            aiPlayer = (player == PLAYER_X) ? PLAYER_O : PLAYER_X;
        }
    }
    
    // Play game against AI
    void playAgainstAI() {
        resetGame();
        
        cout << "\n===== لعبة XO ضد الذكاء الاصطناعي =====\n";
        cout << "أنت: " << humanPlayer << " | الذكاء الاصطناعي: " << aiPlayer << "\n";
        
        while (!gameOver) {
            displayBoard();
            
            if (currentPlayer == humanPlayer) {
                // Human turn
                cout << "دورك! أدخل رقم الخانة (0-8): ";
                int move;
                cin >> move;
                
                if (cin.fail() || !makeMove(move, humanPlayer)) {
                    cin.clear();
                    cin.ignore(numeric_limits<streamsize>::max(), '\n');
                    cout << "حركة غير صحيحة! حاول مرة أخرى.\n";
                    continue;
                }
            } else {
                // AI turn
                cout << "الذكاء الاصطناعي يفكر...\n";
                int move = getAIMove();
                makeMove(move, aiPlayer);
                cout << "الذكاء الاصطناعي اختار الخانة: " << move << "\n";
            }
            
            // Check game status
            char winner = checkWinner();
            if (winner != EMPTY) {
                displayBoard();
                
                if (winner == 'D') {
                    cout << "\n🤝 تعادل!\n";
                    scoreDraw++;
                } else {
                    if (winner == humanPlayer) {
                        cout << "\n🎉 مبروك! لقد فزت!\n";
                    } else {
                        cout << "\n🤖 الذكاء الاصطناعي فاز!\n";
                    }
                    
                    if (winner == PLAYER_X) scoreX++;
                    else scoreO++;
                }
                
                gameOver = true;
                displayScores();
            } else {
                // Switch player
                currentPlayer = (currentPlayer == PLAYER_X) ? PLAYER_O : PLAYER_X;
            }
        }
    }
    
    // Play against friend
    void playAgainstFriend() {
        resetGame();
        
        cout << "\n===== لعبة XO ضد صديق =====\n";
        
        while (!gameOver) {
            displayBoard();
            
            cout << "دور اللاعب " << currentPlayer << "! أدخل رقم الخانة (0-8): ";
            int move;
            cin >> move;
            
            if (cin.fail() || !makeMove(move, currentPlayer)) {
                cin.clear();
                cin.ignore(numeric_limits<streamsize>::max(), '\n');
                cout << "حركة غير صحيحة! حاول مرة أخرى.\n";
                continue;
            }
            
            // Check game status
            char winner = checkWinner();
            if (winner != EMPTY) {
                displayBoard();
                
                if (winner == 'D') {
                    cout << "\n🤝 تعادل!\n";
                    scoreDraw++;
                } else {
                    cout << "\n🏆 اللاعب " << winner << " فاز!\n";
                    if (winner == PLAYER_X) scoreX++;
                    else scoreO++;
                }
                
                gameOver = true;
                displayScores();
            } else {
                // Switch player
                currentPlayer = (currentPlayer == PLAYER_X) ? PLAYER_O : PLAYER_X;
            }
        }
    }
};

// ===== Main Menu =====
void displayMenu() {
    cout << "\n╔════════════════════════════════╗\n";
    cout << "║      لعبة XO - Tic Tac Toe    ║\n";
    cout << "╚════════════════════════════════╝\n";
    cout << "1. اللعب ضد الذكاء الاصطناعي\n";
    cout << "2. اللعب ضد صديق\n";
    cout << "3. تغيير مستوى الصعوبة\n";
    cout << "4. اختيار الرمز (X أو O)\n";
    cout << "5. إعادة النتائج\n";
    cout << "6. الخروج\n";
    cout << "\nاختر (1-6): ";
}

// ===== Main Function =====
int main() {
    TicTacToe game;
    int choice;
    
    cout << "مرحباً بك في لعبة XO!\n";
    
    while (true) {
        displayMenu();
        cin >> choice;
        
        if (cin.fail()) {
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
            cout << "اختيار غير صحيح!\n";
            continue;
        }
        
        switch(choice) {
            case 1:
                game.playAgainstAI();
                break;
            case 2:
                game.playAgainstFriend();
                break;
            case 3: {
                cout << "\nاختر مستوى الصعوبة:\n";
                cout << "1. سهل 😊\n";
                cout << "2. متوسط 😐\n";
                cout << "3. صعب 😈\n";
                cout << "اختر (1-3): ";
                int level;
                cin >> level;
                game.setDifficulty(level);
                cout << "تم تغيير مستوى الصعوبة!\n";
                break;
            }
            case 4: {
                cout << "\nاختر رمزك:\n";
                cout << "1. X\n";
                cout << "2. O\n";
                cout << "اختر (1-2): ";
                int playerChoice;
                cin >> playerChoice;
                game.setHumanPlayer(playerChoice == 1 ? PLAYER_X : PLAYER_O);
                cout << "تم اختيار الرمز!\n";
                break;
            }
            case 5:
                game.resetScores();
                cout << "تم إعادة النتائج!\n";
                break;
            case 6:
                cout << "\nشكراً للعب! وداعاً 👋\n";
                return 0;
            default:
                cout << "اختيار غير صحيح!\n";
        }
    }
    
    return 0;
}

File: game.js
// ===== Game State =====
let gameState = {
    board: Array(9).fill(null),
    currentPlayer: 'X',
    gameMode: 'ai',
    humanPlayer: 'X',
    aiPlayer: 'O',
    difficulty: 'medium',
    isGameOver: false,
    moveCount: 0,
    scores: { X: 0, O: 0, draws: 0 },
    soundEnabled: true
};

// ===== Winning Combinations =====
const WINNING_COMBINATIONS = [
    [0, 1, 2], [3, 4, 5], [6, 7, 8], // Rows
    [0, 3, 6], [1, 4, 7], [2, 5, 8], // Columns
    [0, 4, 8], [2, 4, 6]             // Diagonals
];

// ===== Audio Context for Sound Effects =====
const AudioContext = window.AudioContext || window.webkitAudioContext;
let audioContext;

function initAudio() {
    if (!audioContext) {
        audioContext = new AudioContext();
    }
}

// ===== Sound Effects =====
function playSound(type) {
    if (!gameState.soundEnabled) return;
    
    initAudio();
    
    const oscillator = audioContext.createOscillator();
    const gainNode = audioContext.createGain();
    
    oscillator.connect(gainNode);
    gainNode.connect(audioContext.destination);
    
    switch(type) {
        case 'click':
            oscillator.frequency.value = 800;
            oscillator.type = 'sine';
            gainNode.gain.setValueAtTime(0.3, audioContext.currentTime);
            gainNode.gain.exponentialRampToValueAtTime(0.01, audioContext.currentTime + 0.1);
            oscillator.start(audioContext.currentTime);
            oscillator.stop(audioContext.currentTime + 0.1);
            break;
        case 'win':
            // Victory fanfare
            [523, 659, 784, 1047].forEach((freq, i) => {
                const osc = audioContext.createOscillator();
                const gain = audioContext.createGain();
                osc.connect(gain);
                gain.connect(audioContext.destination);
                osc.frequency.value = freq;
                osc.type = 'sine';
                gain.gain.setValueAtTime(0.3, audioContext.currentTime + i * 0.15);
                gain.gain.exponentialRampToValueAtTime(0.01, audioContext.currentTime + i * 0.15 + 0.3);
                osc.start(audioContext.currentTime + i * 0.15);
                osc.stop(audioContext.currentTime + i * 0.15 + 0.3);
            });
            break;
        case 'draw':
            oscillator.frequency.value = 400;
            oscillator.type = 'triangle';
            gainNode.gain.setValueAtTime(0.2, audioContext.currentTime);
            gainNode.gain.exponentialRampToValueAtTime(0.01, audioContext.currentTime + 0.3);
            oscillator.start(audioContext.currentTime);
            oscillator.stop(audioContext.currentTime + 0.3);
            break;
        case 'reset':
            oscillator.frequency.value = 600;
            oscillator.type = 'square';
            gainNode.gain.setValueAtTime(0.2, audioContext.currentTime);
            gainNode.gain.exponentialRampToValueAtTime(0.01, audioContext.currentTime + 0.15);
            oscillator.start(audioContext.currentTime);
            oscillator.stop(audioContext.currentTime + 0.15);
            break;
    }
}

// ===== Theme Toggle =====
function toggleTheme() {
    const body = document.body;
    body.classList.toggle('dark-mode');
    body.classList.toggle('light-mode');
    playSound('click');
    localStorage.setItem('theme', body.classList.contains('dark-mode') ? 'dark' : 'light');
}

// Load saved theme
window.addEventListener('DOMContentLoaded', () => {
    const savedTheme = localStorage.getItem('theme') || 'light';
    document.body.className = savedTheme === 'dark' ? 'dark-mode' : 'light-mode';
    loadScores();
});

// ===== Sound Toggle =====
function toggleSound() {
    gameState.soundEnabled = !gameState.soundEnabled;
    document.body.classList.toggle('sound-muted');
    if (gameState.soundEnabled) {
        playSound('click');
    }
}

// ===== Mode Selection =====
function selectMode(mode) {
    gameState.gameMode = mode;
    
    document.querySelectorAll('.mode-btn').forEach(btn => {
        btn.classList.remove('active');
    });
    document.querySelector(`[data-mode="${mode}"]`).classList.add('active');
    
    const playerSelection = document.getElementById('playerSelection');
    const difficultySelection = document.getElementById('difficultySelection');
    
    if (mode === 'ai') {
        playerSelection.style.display = 'block';
        difficultySelection.style.display = 'block';
        updateLabels();
    } else {
        playerSelection.style.display = 'none';
        difficultySelection.style.display = 'none';
        document.getElementById('labelX').textContent = 'لاعب X';
        document.getElementById('labelO').textContent = 'لاعب O';
    }
    
    playSound('click');
    restartGame();
}

// ===== Player Selection =====
function selectPlayer(player) {
    gameState.humanPlayer = player;
    gameState.aiPlayer = player === 'X' ? 'O' : 'X';
    
    document.querySelectorAll('.player-btn').forEach(btn => {
        btn.classList.remove('active');
    });
    document.querySelector(`[data-player="${player}"]`).classList.add('active');
    
    updateLabels();
    playSound('click');
    restartGame();
}

function updateLabels() {
    if (gameState.gameMode === 'ai') {
        document.getElementById('labelX').textContent = gameState.humanPlayer === 'X' ? 'أنت' : 'الذكاء الاصطناعي';
        document.getElementById('labelO').textContent = gameState.aiPlayer === 'O' ? 'الذكاء الاصطناعي' : 'أنت';
    }
}

// ===== Difficulty Selection =====
function selectDifficulty(level) {
    gameState.difficulty = level;
    
    document.querySelectorAll('.diff-btn').forEach(btn => {
        btn.classList.remove('active');
    });
    document.querySelector(`[data-level="${level}"]`).classList.add('active');
    
    playSound('click');
    restartGame();
}

// ===== Cell Click Handler =====
function handleCellClick(index) {
    if (gameState.board[index] || gameState.isGameOver) return;
    
    // In AI mode, prevent clicking when it's AI's turn
    if (gameState.gameMode === 'ai' && gameState.currentPlayer !== gameState.humanPlayer) return;
    
    makeMove(index, gameState.currentPlayer);
}

// ===== Make Move =====
function makeMove(index, player) {
    console.log('Making move:', index, 'for player:', player);
    
    gameState.board[index] = player;
    gameState.moveCount++;
    
    const cell = document.querySelector(`[data-index="${index}"]`);
    cell.textContent = player;
    cell.classList.add('taken', player.toLowerCase());
    
    playSound('click');
    updateMoveCount();
    
    const result = checkGameOver();
    
    if (result) {
        handleGameOver(result);
    } else {
        gameState.currentPlayer = gameState.currentPlayer === 'X' ? 'O' : 'X';
        updateTurnIndicator();
        
        console.log('Next player:', gameState.currentPlayer, 'Game mode:', gameState.gameMode, 'AI player:', gameState.aiPlayer);
        
        // AI's turn
        if (gameState.gameMode === 'ai' && gameState.currentPlayer === gameState.aiPlayer) {
            console.log('Scheduling AI move...');
            setTimeout(makeAIMove, 300);
        }
    }
}

// ===== AI Move =====
function makeAIMove() {
    if (gameState.isGameOver) return;
    
    console.log('AI is making a move...', 'Current player:', gameState.currentPlayer, 'AI player:', gameState.aiPlayer);
    
    let move;
    
    switch(gameState.difficulty) {
        case 'easy':
            move = getRandomMove();
            break;
        case 'medium':
            move = Math.random() > 0.5 ? getBestMove() : getRandomMove();
            break;
        case 'hard':
            move = getBestMove();
            break;
        default:
            move = getRandomMove();
    }
    
    console.log('AI chose move:', move);
    
    if (move !== -1) {
        makeMove(move, gameState.aiPlayer);
    }
}

// ===== Random Move (Easy AI) =====
function getRandomMove() {
    const availableMoves = gameState.board
        .map((cell, index) => cell === null ? index : -1)
        .filter(index => index !== -1);
    
    return availableMoves.length > 0 
        ? availableMoves[Math.floor(Math.random() * availableMoves.length)]
        : -1;
}

// ===== Minimax Algorithm (Hard AI) =====
function minimax(board, depth, isMaximizing, alpha = -Infinity, beta = Infinity) {
    const result = checkWinnerForBoard(board);
    
    if (result === gameState.aiPlayer) return 10 - depth;
    if (result === gameState.humanPlayer) return depth - 10;
    if (result === 'draw') return 0;
    
    if (isMaximizing) {
        let bestScore = -Infinity;
        for (let i = 0; i < 9; i++) {
            if (!board[i]) {
                board[i] = gameState.aiPlayer;
                const score = minimax(board, depth + 1, false, alpha, beta);
                board[i] = null;
                bestScore = Math.max(score, bestScore);
                alpha = Math.max(alpha, bestScore);
                if (beta <= alpha) break; // Alpha-Beta Pruning
            }
        }
        return bestScore;
    } else {
        let bestScore = Infinity;
        for (let i = 0; i < 9; i++) {
            if (!board[i]) {
                board[i] = gameState.humanPlayer;
                const score = minimax(board, depth + 1, true, alpha, beta);
                board[i] = null;
                bestScore = Math.min(score, bestScore);
                beta = Math.min(beta, bestScore);
                if (beta <= alpha) break; // Alpha-Beta Pruning
            }
        }
        return bestScore;
    }
}

function getBestMove() {
    let bestScore = -Infinity;
    let bestMove = -1;
    
    for (let i = 0; i < 9; i++) {
        if (!gameState.board[i]) {
            gameState.board[i] = gameState.aiPlayer;
            const score = minimax([...gameState.board], 0, false);
            gameState.board[i] = null;
            
            if (score > bestScore) {
                bestScore = score;
                bestMove = i;
            }
        }
    }
    
    return bestMove;
}

// ===== Check Winner =====
function checkWinnerForBoard(board) {
    for (const combo of WINNING_COMBINATIONS) {
        const [a, b, c] = combo;
        if (board[a] && board[a] === board[b] && board[a] === board[c]) {
            return board[a];
        }
    }
    
    if (board.every(cell => cell !== null)) {
        return 'draw';
    }
    
    return null;
}

function checkGameOver() {
    const winner = checkWinnerForBoard(gameState.board);
    
    if (winner && winner !== 'draw') {
        const winningCombo = WINNING_COMBINATIONS.find(combo => {
            const [a, b, c] = combo;
            return gameState.board[a] && 
                   gameState.board[a] === gameState.board[b] && 
                   gameState.board[a] === gameState.board[c];
        });
        
        return { type: 'win', player: winner, combo: winningCombo };
    }
    
    if (winner === 'draw') {
        return { type: 'draw' };
    }
    
    return null;
}

// ===== Handle Game Over =====
function handleGameOver(result) {
    gameState.isGameOver = true;
    
    if (result.type === 'win') {
        // Highlight winning cells
        result.combo.forEach(index => {
            document.querySelector(`[data-index="${index}"]`).classList.add('winner');
        });
        
        gameState.scores[result.player]++;
        playSound('win');
        
        let message;
        if (gameState.gameMode === 'ai') {
            message = result.player === gameState.humanPlayer 
                ? '🎉 مبروك! لقد فزت!' 
                : '🤖 الذكاء الاصطناعي فاز!';
        } else {
            message = `اللاعب ${result.player} فاز!`;
        }
        
        showModal('🏆', 'فوز!', message);
    } else {
        gameState.scores.draws++;
        playSound('draw');
        showModal('🤝', 'تعادل!', 'اللعبة انتهت بالتعادل');
    }
    
    updateScoreboard();
    saveScores();
}

// ===== Modal =====
function showModal(icon, title, message) {
    document.getElementById('modalIcon').textContent = icon;
    document.getElementById('modalTitle').textContent = title;
    document.getElementById('modalMessage').textContent = message;
    document.getElementById('gameOverModal').classList.add('show');
}

function closeModal() {
    document.getElementById('gameOverModal').classList.remove('show');
}

function playAgain() {
    closeModal();
    restartGame();
}

// ===== Update UI =====
function updateTurnIndicator() {
    const turnElement = document.getElementById('currentTurn');
    turnElement.textContent = gameState.currentPlayer;
    
    if (gameState.gameMode === 'ai' && gameState.currentPlayer === gameState.aiPlayer) {
        document.getElementById('turnIndicator').innerHTML = 
            '<span class="turn-text">🤖 الذكاء الاصطناعي يفكر...</span>';
    } else {
        document.getElementById('turnIndicator').innerHTML = 
            `<span class="turn-text">الدور:</span><span class="turn-player">${gameState.currentPlayer}</span>`;
    }
}

function updateMoveCount() {
    document.getElementById('moveCount').textContent = gameState.moveCount;
}

function updateScoreboard() {
    document.getElementById('scoreX').textContent = gameState.scores.X;
    document.getElementById('scoreO').textContent = gameState.scores.O;
    document.getElementById('scoreDraw').textContent = gameState.scores.draws;
}

// ===== Game Controls =====
function restartGame() {
    gameState.board = Array(9).fill(null);
    gameState.currentPlayer = 'X';
    gameState.isGameOver = false;
    gameState.moveCount = 0;
    
    document.querySelectorAll('.cell').forEach(cell => {
        cell.textContent = '';
        cell.className = 'cell';
    });
    
    updateTurnIndicator();
    updateMoveCount();
    closeModal();
    playSound('reset');
    
    // If AI starts first
    if (gameState.gameMode === 'ai' && gameState.aiPlayer === 'X') {
        setTimeout(makeAIMove, 300);
    }
}

function resetScores() {
    if (confirm('هل أنت متأكد من إعادة تعيين جميع النتائج؟')) {
        gameState.scores = { X: 0, O: 0, draws: 0 };
        updateScoreboard();
        saveScores();
        restartGame();
        playSound('reset');
    }
}

// ===== Local Storage =====
function saveScores() {
    localStorage.setItem('xoScores', JSON.stringify(gameState.scores));
}

function loadScores() {
    const saved = localStorage.getItem('xoScores');
    if (saved) {
        gameState.scores = JSON.parse(saved);
        updateScoreboard();
    }
}

// ===== Initialize Game =====
updateTurnIndicator();
updateMoveCount();
updateScoreboard();
updateLabels();
